warehouse/integrations/bigquery/bigquery_test.go

```diff
@@ -353,6 +352,12 @@ func TestIntegration(t *testing.T) {
 	})
 
 	t.Run("Validations", func(t *testing.T) {
+		t.Cleanup(func() {
+			require.NoError(t, testhelper.WithConstantRetries(func() error {
```

Comment:
I don't like much the constant retries helper, I suggest we move to `require.Eventually`. The reason I'm saying this is because with `Eventually` we can set both a total timeout and a retry interval. By doing this it is not just more readable and intuitive, it usuallly ends up retrying faster leading to a test finishing faster.

Reply:
Added the changes.